src/braket/circuits/text_diagram_builders/unicode_circuit_diagram.py

```diff
@@ -0,0 +1,282 @@
+        return 1
+
+    @classmethod
+    def _duplicate_time_at_bottom(cls, lines: list) -> None:
+        # Do not add a line after the circuit
+        # It is safe to do because the last line is empty: _qubit_line_spacing["after"] = 1
+        lines[-1] = lines[0]
+
+    @classmethod
+    def _create_diagram_column(
+        cls,
+        circuit_qubits: QubitSet,
+        items: list[Instruction | ResultType],
+        global_phase: float | None = None,
+    ) -> str:
+        """Return a column in the string diagram of the circuit for a given list of items.
+
+        Args:
+            circuit_qubits (QubitSet): qubits in circuit
+            items (list[Instruction | ResultType]): list of instructions or result types
+            global_phase (float | None): the integrated global phase up to this column
+
+        Returns:
+            str: a string diagram for the specified moment in time for a column.
+        """
+        symbols = {qubit: cls._qubit_line_character() for qubit in circuit_qubits}
+        connections = {qubit: "none" for qubit in circuit_qubits}
+
+        for item in items:
+            (
+                target_qubits,
+                control_qubits,
+                qubits,
+                connections,
+                ascii_symbols,
+                map_control_qubit_states,
+            ) = cls._build_parameters(circuit_qubits, item, connections)
+
+            for qubit in qubits:
+                # Determine if the qubit is part of the item or in the middle of a
+                # multi qubit item.
+                if qubit in target_qubits:
+                    item_qubit_index = [
+                        index for index, q in enumerate(target_qubits) if q == qubit
+                    ][0]
+                    power_string = (
+                        f"^{power}"
+                        if (
+                            (power := getattr(item, "power", 1)) != 1
+                            # this has the limitation of not printing the power
+                            # when a user has a gate genuinely named C, but
+                            # is necessary to enable proper printing of custom
+                            # gates with built-in control qubits
+                            and ascii_symbols[item_qubit_index] != "C"
+                        )
+                        else ""
+                    )
+                    symbols[qubit] = (
+                        f"{ascii_symbols[item_qubit_index]}{power_string}"
+                        if power_string
+                        else ascii_symbols[item_qubit_index]
+                    )
+
+                elif qubit in control_qubits:
+                    symbols[qubit] = "C" if map_control_qubit_states[qubit] else "N"
+                else:
+                    symbols[qubit] = "┼"
+
+        output = cls._create_output(symbols, connections, circuit_qubits, global_phase)
+        return output
+
+    @staticmethod
```

Comment:
nit: we could have made this a class method, rather than referencing the class inside it.